hcl/hclsyntax/parser.go

```diff
@@ -236,10 +236,18 @@ func (p *parser) finishParsingBodyAttribute(ident Token, singleLine bool) (Node,
 			end := p.Peek()
 			if end.Type != TokenNewline && end.Type != TokenEOF {
 				if !p.recovery {
+					summary := "Missing newline after argument"
+					detail := "An argument definition must end with a newline."
+
+					if end.Type == TokenComma {
+						summary = "Unexpected comma after argument"
+						detail = "Comma-separated argument definitions are no longer allowed. " + detail
```

Comment:
This seems reasonable to me, but I'd prefer to say something like "Argument definitions must be separated by newlines, not commas."

We've kept away from phrasing like "no longer" because such messages won't age well over time when (hopefully) many users never saw a language where it _was_ allowed.

Reply:
Good call.